packages/wdio-utils/src/test-framework/testFnWrapper.ts

```diff
@@ -66,8 +87,8 @@ export const testFrameworkFnWrapper = async function (
      * (in Jasmine failing assertions are not causing the test to throw as
      * oppose to other common assertion libraries like chai)
      */
-    if (!error && afterArgs[0] && afterArgs[0].failedExpectations && afterArgs[0].failedExpectations.length) {
-        error = afterArgs[0].failedExpectations[0]
+    if (!error && afterArgs[0] && (afterArgs as [JasmineContext, unknown])[0].failedExpectations && (afterArgs as [JasmineContext, unknown])[0].failedExpectations.length) {
```

Comment:
on line 83, change it to: `let afterArgs = afterFnArgs(this) as [JasmineContext, unknown]` so that you can remove all this castings here

Reply:
this if statement is specific to jasmine which is why I have to case it only here